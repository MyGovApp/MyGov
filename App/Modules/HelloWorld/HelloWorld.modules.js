import { handleAction, createReducer } from '../../Utilities/handleAction'

export const toggleColor = () => ({ type: 'TOGGLE_COLOR' })

handleAction('TOGGLE_COLOR', (state, action) => ({
  color: state.color === 'red' ? 'blue' : 'red'
}))

const initialState = {
  color: 'red'
}

export default createReducer(initialState)
