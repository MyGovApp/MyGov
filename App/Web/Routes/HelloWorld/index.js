import { injectReducer } from '../../Store/reducers'

export default (store) => ({
  path: 'hello-world',
  getComponent (nextState, next) {
    require.ensure([
      './helloWorld.container',
      '../../../Modules/HelloWorld/HelloWorld.modules'
    ], (require) => {
      const helloWorld = require('./helloWorld.container').default
      const helloWorldReducer = require('../../../Modules/HelloWorld/HelloWorld.modules').default

      injectReducer(store, {
        key: 'helloWorld',
        reducer: helloWorldReducer
      })

      next(null, helloWorld)
    })
  }
})
