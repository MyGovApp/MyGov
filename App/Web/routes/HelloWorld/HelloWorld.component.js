import React, { Component, PropTypes } from 'react'
import classes from './HelloWorld.styles.scss'

class HelloWorld extends Component {
  constructor () {
    super()
    this.state = {}
  }
  render () {
    const { toggleColor, color } = this.props
    return (
      <div className={classes.container}>
        <h2
          title='helloWorld'
          onClick={toggleColor}
          className={classes.text}
          style={{ color }}>
          Hello World!
        </h2>
      </div>
    )
  }
}

HelloWorld.propTypes = {
  color: PropTypes.string.isRequired,
  toggleColor: PropTypes.func.isRequired
}

module.exports = HelloWorld
