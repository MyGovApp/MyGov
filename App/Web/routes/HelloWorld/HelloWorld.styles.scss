.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.text {
  cursor: pointer;
}
