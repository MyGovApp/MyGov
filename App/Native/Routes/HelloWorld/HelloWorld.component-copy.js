import React, { Component } from 'react'
import { View, Text } from 'react-native'
import { classes } from './HelloWorld.styles'

export default class HelloWorld extends Component {
  render () {
    return (
      <View style={classes.mainView}>
        <Text>
          Hello World!
        </Text>
      </View>
    )
  }
}
