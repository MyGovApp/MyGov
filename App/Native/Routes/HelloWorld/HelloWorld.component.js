import React, { Component, PropTypes } from 'react'
import { View, Text } from 'react-native'
import { classes } from './HelloWorld.styles'

export default class HelloWorld extends Component {
  render () {
    const { toggleColor, color } = this.props

    return (
      <View style={classes.mainView}>
        <Text
          onPress={toggleColor}
          style={[classes.helloWorld, { color }]}>
          Hello World!
        </Text>
      </View>
    )
  }
}

HelloWorld.propTypes = {
  color: PropTypes.string.isRequired,
  toggleColor: PropTypes.func.isRequired
}
