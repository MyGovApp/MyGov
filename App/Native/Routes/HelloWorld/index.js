import HelloWorld from './HelloWorld.container.js'

export default HelloWorld
