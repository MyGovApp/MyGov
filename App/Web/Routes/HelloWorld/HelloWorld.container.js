import { connect } from 'react-redux'
import { toggleColor } from '../../../Modules/HelloWorld/HelloWorld.modules'
import HelloWorld from './HelloWorld.component'

const mapDispatchToProps = {
  toggleColor
}

const mapStateToProps = (state, ownProps) => ({
  color: state.helloWorld.color
})

export default connect(mapStateToProps, mapDispatchToProps)(HelloWorld)
