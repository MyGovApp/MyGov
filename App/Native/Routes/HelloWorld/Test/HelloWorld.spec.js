import 'react-native'
import React from 'react' // eslint-disable-line
import renderer from 'react-test-renderer'
import { ComponentWrapper, ContainerWrapper } from './HelloWorld.setup'

describe('Native Hello World', () => {
  it('should render the HelloWorld component correctly', () => {
    const tree = renderer.create(ComponentWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render the HelloWorld container correctly', () => {
    const tree = renderer.create(ContainerWrapper()).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
