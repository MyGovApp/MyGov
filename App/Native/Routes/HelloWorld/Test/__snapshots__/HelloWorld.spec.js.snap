// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Native Hello World should render the HelloWorld component correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Hello World!
  </Text>
</View>
`;

exports[`Native Hello World should render the HelloWorld container correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "fontFamily": "HelveticaNeue",
          "fontSize": 20,
          "textAlign": "center",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
    Hello World!
  </Text>
</View>
`;
